src/Microsoft.TemplateEngine.Cli/HelpAndUsage/HelpForTemplateResolution.cs

```diff
@@ -50,6 +51,48 @@ public static CreationResultStatus CoordinateHelpAndUsageDisplay(TemplateListRes
             }
         }
 
+        /// <summary>
+        /// Displays the help in case it is not possible to resolve template to use based on user input on template instantiation.
+        /// </summary>
+        /// <param name="resolutionResult">template resolution result</param>
+        /// <param name="environmentSettings"></param>
+        /// <param name="commandInput">command input used in CLI</param>
+        /// <param name="defaultLanguage">default language for the host</param>
+        /// <param name="installUnitDescriptors">the collection of install unit descriptors</param>
+        /// <returns></returns>
+        internal static CreationResultStatus CoordinateAmbiguousTemplateResolutionDisplay(
+            TemplateResolutionResult resolutionResult,
+            IEngineEnvironmentSettings environmentSettings,
+            INewCommandInput commandInput,
+            string defaultLanguage,
+            IEnumerable<IInstallUnitDescriptor> installUnitDescriptors)
+        {
+            switch (resolutionResult.ResolutionStatus)
+            {
+                case TemplateResolutionResult.Status.NoMatch:
+                    Reporter.Error.WriteLine(
+                        string.Format(LocalizableStrings.NoTemplatesMatchingInputParameters, GetInputParametersString(commandInput)).Bold().Red());
+                    Reporter.Error.WriteLine(LocalizableStrings.ListTemplatesCommand.Bold().Red());
+                    return CreationResultStatus.NotFound;
+                case TemplateResolutionResult.Status.AmbiguousLanguageChoice:
```

Comment:
I think we want to split `case TemplateResolutionResult.Status.AmbiguousLanguageChoice:` into its own message.